### PR TITLE
Re-add pump time check

### PR DESCRIPTION
**Pull request by jotomo**

### CODE REVIEW

File: app/src/main/java/info/nightscout/androidaps/plugins/PumpCombo/ComboPlugin.java [hunk 1]
```diff
@@ -727,10 +726,20 @@ private CommandResult runOnConnectChecks() {
...
         checkAndResolveTbrMismatch(preCheckResult.state);
+        checkPumpTime(preCheckResult.state);
 
         return null;
     }
 
+    /** check pump time (main menu) and  raise notification if clock is off by more than 2m
+     * (setting clock is not supported by ruffy) */
+    private void checkPumpTime(PumpState state) {
+        if (state.timestamp != 0 && Math.abs(state.timestamp - System.currentTimeMillis()) > 2 * 60 * 1000) {
```
Comment (AdrianLxM) [inline app/src/main/java/info/nightscout/androidaps/plugins/PumpCombo/ComboPlugin.java]: This checks for `state.timestamp`, the last connection time, not for `state.pumpTime`.
It alarms with those changes: https://github.com/jotomo/AndroidAPS/pull/7

**Comment by jotomo** [comment]
@AdrianLxM Urgency level is normal. Should we make this critical when time is off not by just a few minutes but an hour or more (screwing up basal rates considerably)? E.g. after a battery change the clock might reset.

**Comment by AdrianLxM** [comment]
@jotomo sounds like a good idea. What about reading the last bolus after a bolus was cancelled or interrupted? How precise must the timing be there to detect the actual amount delivered?

**Comment by jotomo** [comment]
I've updated the BolusCommand to check the last bolus only bolus delivery was actually started. The time isn't checked anymore (due to imprecision). But with the time-check added again, should this be checked again?

**Comment by AdrianLxM** [comment]
So how about: 3 to 9:59 minutes off: Notification
10 Minutes or more off: Urgent Notification.

If you changed the bolus checks like this, a time check would only be for extra safety. Imho it would be ok to be within "not yet urgent notification" granularity - only to check that date changes may not have messed with the history (so far I haven't seen that in a combo in RT mode though).

**Comment by jotomo** [comment]
Updated and tested to trigger with different urgency depending on size of deviation.

**Comment by jotomo** [comment]
Merged locally.

**Pull request closed without merging**